source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -792,13 +792,18 @@ namespace
         {
             RtAttribute attr = node.getPrim().getAttribute(attrDef.getName());
             RtInput input = attr.asA<RtInput>();
-            if (input)
+            if (input && attrDef)
```

Comment:
The null/valid check on attrDef seems unnecessary here. You are already accessing it above with `attrDef.getName()`.

I don't think the iterator return from getAttributes() could return an invalid RtAttribute. Unless you have found that to happen in some circumstances?

Reply:
Ah, ok, I'll remove it.